test/x/media_uploader_test.rb

```diff
@@ -68,6 +68,26 @@ def test_await_processing_failed
       assert_equal "failed", response["processing_info"]["state"]
     end
 
+    def test_await_processing!
+      stub_request(:get, "https://api.twitter.com/2/media/upload?command=STATUS&media_id=#{TEST_MEDIA_ID}")
+        .to_return(headers: {"content-type" => "application/json"}, body: '{"data":{"processing_info": {"state": "pending"}}}')
+        .to_return(headers: {"content-type" => "application/json"}, body: '{"data":{"processing_info": {"state": "succeeded"}}}')
+
+      result = MediaUploader.await_processing!(client: @client, media: MEDIA)
+
+      assert_equal "succeeded", result["processing_info"]["state"]
+    end
+
+    def test_await_processing_and_failed!
```

Comment:
Minor feedback: I would name this method `test_await_processing_failed!` (removing `_and_`) to match the `test_await_processing_failed` method (without the exclamation point) above.